pkg/provisioner/provisioner.go

```diff
@@ -18,9 +18,18 @@ type EventPublisher func(reason, message string)
 // Factory is the interface for creating new Provisioner objects.
 type Factory func(host *metal3v1alpha1.BareMetalHost, bmcCreds bmc.Credentials, publish EventPublisher) (Provisioner, error)
 
-// UserDataSource is the interface for a function to retrieve user
-// data for a host being provisioned.
-type UserDataSource func() (string, error)
+// HostConfigData retrieves host configuration data
+type HostConfigData interface {
+	// UserData is the interface for a function to retrieve user
+	// data for a host being provisioned.
+	UserData() (string, error)
+
+	// NetworkData is the interface for a function to retrieve netwok
+	// configuration for a host.
+	NetworkData() ([]byte, error)
```

Comment:
Why do we expect a byte array here but a string for user data?

Reply:
ConfigDrive object from gophercloud library interprets NetworkData as map[string]interface{} which requires 'unmarshalling' data from secret therefore network data has to be []byte. That being said we can convert everything to string (for the price of yet another string -> []byte conversion)  fir the sake of consistency